### PR TITLE
show mod name when placing a block

### PR DESCRIPTION
**Pull request by DeltaNedas**
not foolproof
will not show for existing blocks (idk how)
[See video of it!](https://streamable.com/hqn5xl)
just a note for mods without color in their displayNames, default is [accent] as to differentiate between item name and mod name

### CODE REVIEW

**Comment by DeltaNedas** [comment]
just a note for mods without color in their displayNames, default is [accent]

**Comment by Quezler** [comment]
video link appears broken

**Comment by DeltaNedas** [comment]
works for me in incognito and normal window, stop phoneposting
i put it on discord too anyways

**Comment by DeltaNedas** [comment]
ok yea its broken now does it expire?

File: core/src/mindustry/ui/fragments/PlacementFragment.java [hunk 1]
```diff
@@ -271,9 +272,18 @@ public void build(Group parent){
                                     }
                                 }
                                 final String keyComboFinal = keyCombo;
+                                String modName = "";
+                                for(LoadedMod mod : mods.list()){
```
Comment (Anuken) [inline core/src/mindustry/ui/fragments/PlacementFragment.java]: You can use `modNameFinal = lastDisplay.minfo.mod != null ? lastDisplay.minfo.mod.meta.displayName : ""`

Comment (DeltaNedas) [reply core/src/mindustry/ui/fragments/PlacementFragment.java]: thank you anuke, i had no idea blocks stored their mod info!
hell ill try and add it for placed blocks too, if i can find its fragment

**Comment by DeltaNedas** [comment]
now to try add it for placed blocks (pls no merge just yet)

File: core/src/mindustry/world/Block.java [hunk 2]
```diff
@@ -671,6 +671,14 @@ public String getDisplayName(Tile tile){
         return localizedName;
     }
 
+    public String getModName(String prefix){
```
Comment (Anuken) [inline core/src/mindustry/world/Block.java]: These method should be in `Content`, since they're not block-specific.

Comment (DeltaNedas) [reply core/src/mindustry/world/Block.java]: youre right ill move it now

**Comment by DeltaNedas** [comment]
oh no
why did i do that

**Comment by DeltaNedas** [comment]
what
it builds fine for me

**Comment by DeltaNedas** [comment]
hey anuke you may want to squash commits...

**Comment by Anuken** [comment]
I took a look at how this appeared in-game, and the mod name label just takes up a lot of space. It interferes with the alignment of the icon.

Instead of displaying mod names in-game, I decided to make it display them only in the tech tree and database. This makes it so that you can see the mod for *all* content, and keeps the placement UI clean.

**Comment by DeltaNedas** [comment]
good call

**Pull request closed without merging**